handler/smc_handler.py

```diff
@@ -91,44 +91,58 @@ def sender_address(self):
     def config(self):
         return self._config
 
+    @property
+    def basic_call_context(self):
+        return make_call_context(
+            sender_address=self.sender_address,
+            gas=self.config["DEFAULT_GAS"]
+        )
     #
     # Public variable getter functions
     #
-    def get_eligible_proposer(self, shard_id, period=None, gas=None):
+
+    def does_notary_exist(self, notary_address):
+        return self.functions.does_notary_exist(notary_address).call(self.basic_call_context)
+
+    def get_notary_info(self, notary_address):
+        return self.functions.get_notary_info(notary_address).call(self.basic_call_context)
+
+    def notary_pool_len(self):
+        return self.functions.notary_pool_len().call(self.basic_call_context)
+
+    def empty_slots_stack_top(self):
+        return self.functions.empty_slots_stack_top().call(self.basic_call_context)
+
+    def empty_slots_stack(self):
+        return self.functions.empty_slots_stack().call(self.basic_call_context)
```

Comment:
Are the stack `empty_slots_stack` things for tests only? IMO I don't think they should be accessed directly.